### ISSUE TITLE
Build failure using v6.22.0 / uses unsafe build flags

### ISSUE THREAD

**Issue by maxcleme** [bugsnag/bugsnag-cocoa#1456]
### Describe the bug

When building our application with v6.22.0, it fails:

`The package product 'Bugsnag' cannot be used as a dependency of this target because it uses unsafe build flags.`

Maybe due to this ?

https://github.com/bugsnag/bugsnag-cocoa/pull/1455/files#diff-c4c40d3e49de76d8fac0fa28120520e20f1131fbca42210efb854a3dde7bb105R41

### Steps to reproduce
1. Use v6.22.0 as exact version
2. Try to build

### Environment
* Bugsnag version: v6.22.0
* CocoaPods version: n/a
* Carthage version (if any): n/a
* iOS/tvOS/macOS version(s): 12.5
* Simulator or physical device: Physical, M1, 2020
* Xcode version: 13.3 (13E113)
* Swift version (if applicable): 5

### Example code snippet 

```
{
	isa = XCRemoteSwiftPackageReference;
	repositoryURL = "https://github.com/bugsnag/bugsnag-cocoa";
	requirement = {
		kind = exactVersion;
		version = 6.22.0;
	};
};
```

<details><summary>Error messages:</summary>

```
The package product 'Bugsnag' cannot be used as a dependency of this target because it uses unsafe build flags.
```
</details>

**Comment by kvyatkovskys** [comment]
We have the same issue

**Comment by luke-belton** [comment]
Hi @maxcleme/@kvyatkovskys - thanks for raising this. We're investigating this now and will keep this thread updated 👍

**Comment by luke-belton** [comment]
Hi all - just to let you know that a [fix](https://github.com/bugsnag/bugsnag-cocoa/pull/1457) for this was released in [`v6.22.1`](https://github.com/bugsnag/bugsnag-cocoa/releases/tag/6.22.1) 🎉

**Comment by kvyatkovskys** [comment]
> Hi all - just to let you know that a [fix](https://github.com/bugsnag/bugsnag-cocoa/pull/1457) for this was released in [`v6.22.1`](https://github.com/bugsnag/bugsnag-cocoa/releases/tag/6.22.1) 🎉

thank you 🍻
